### PR TITLE
Doxygen 1.9.1

### PR DESCRIPTION
**Pull request by datalogics-robb**
Specify library name and version:  **doxygen/1.9.1**

Doxygen needs a bit of patching to build on AIX, primarily because the OS headers seem to be a little sloppier than we're used to.
This PR will require the changes in [PR#5151](https://github.com/conan-io/conan-center-index/pull/5151) to get the dependency `xapian-core`, that also requires modifications to build on AIX.
The patch moves some include statements, as the ordering broke things on AIX, specifically the definition of the `PRIu64` format specifier.
AIX also seemed to have definitions for TRUE and FALSE in system headers that conflicted with the use in Doxygen.
Doxygen 1.9.1 did NOT require the CMake file patching performed on the 1.8 release series, so that was not applied to 1.9.1

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/doxygen/all/conanfile.py [hunk 1]
```diff
@@ -77,6 +77,10 @@ def package(self):
         cmake = self._configure_cmake()
         cmake.install()
 
+    def package_id(self):
+        self.info.include_build_settings()
+        del self.info.settings.compiler
```
Comment (madebr) [inline recipes/doxygen/all/conanfile.py:82]:
```suggestion
        del self.info.settings.build_type
        del self.info.settings.compiler
```

- We don't need to build a Debug doxygen executable.
- I don't think there is a need to do `self.info.include_build_settings()`. `arch_build` and `os_build` are not included in `settings`.
See https://github.com/conan-io/conan/blob/f230bf104842c160be6ea377497151fe1ea64c38/conans/model/info.py#L628-L630

**Comment by datalogics-robb** [comment]
Now that  PR#5151 has been merged, I rebased this branch on a current upstream, and force-pushed to trigger CI.

**Comment by datalogics-robb** [comment]
Ok, problems  with a bad push were resolved, and this version was used to build packages for
Macos x86_64
Linux x86_64
Windows x86_64
Linux AArch64 (armv8)
Solaris SPARCv9
AIX ppc64

**Comment by datalogics-robb** [comment]
Grr, it won't let me re-open because the branch was force-pushed or recreated.

**Comment by madebr** [comment]
Thanks!
Let's all head over to #5201

**Pull request closed without merging**